colcon_sanitizer_reports/sanitizer_log_parser.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import defaultdict
+import csv
+from io import StringIO
+import re
+from typing import Dict, List, NamedTuple, Optional
+
+from colcon_sanitizer_reports._sanitizer_section import SanitizerSection
+
+# The start line of a section can be found with the following regex. Additionally, any prefix that
+# is prepended by the logging system can be extracted and be used to lstrip following section lines.
+_FIND_SECTION_START_LINE_REGEX = \
+    re.compile(r'^(?P<prefix>.*?)(==\d+==|)(WARNING|ERROR):.*Sanitizer:.*$')
+
+# The end line of a section can be found with the following regex. Additionally, any prefix that is
+# prepended by the logging system can be extracted and be used to match the common prefix of the
+# section to which the end line belongs.
+_FIND_SECTION_END_LINE_REGEX = re.compile(r'^(?P<prefix>.*?)(SUMMARY: .*Sanitizer: .*)$')
```

Comment:
Same question as above regarding `.*?`.

Reply:
This could just be .* and not .*? in this case. There's no need for a non-greedy match. Will change.